ethereum/core/src/main/java/tech/pegasys/teku/core/ForkChoiceUtil.java

```diff
@@ -399,6 +402,11 @@ private static AttestationProcessingResult indexAndValidateAttestation(
         .ifSuccessful(
             () -> {
               attestation.setIndexedAttestation(indexedAttestation);
+
+              if (attestation.getMaybeIndexedAttestation().isEmpty()) {
+                attestation.saveRandaoMix(targetState);
+              }
```

Comment:
It may be cleaner to just call `attestation.saveRandoMix` here and have the method do nothing if it already has a randaoMix saved.  Otherwise it takes a bit of reasoning to understand why the absence of an indexed attestation means we need to save randao mix.

Reply:
My bad, I was actually trying to call `attestation.getMaybeRandaoMix().isEmpty()

Reply:
But your suggestion below makes sense. I'll apply that one.

Reply:
Done.